.github/workflows/deploy.yml

```diff
@@ -0,0 +1,44 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - examples/**
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - name: Install and build 🔧
+        run: |
+          yarn install --frozen-lockfile
+          yarn build
+
+      - name: Build examples
+        run: |
+          yarn examples:install
+          yarn examples:build
+
+      - name: Delete files not needed for production
+        run: |
+          rm -rf ./examples/*/node_modules
```

Comment:
Maybe this should be done via some script and run here only `yarn examples:cleanup`.

Reply:
The deletion is here only for the purpose of deployment and it's not meant to be run on developer's machine, it deletes files that are under version control. If `JamesIves/github-pages-deploy-action` supported some kind of `ignore` or `exclude`, we could leave this off. Therefore I think it's OK the way it is.

Reply:
Alternatively, we could prepare examples in a safe way, just like we do with `dist` — hand picked and excluded from version control. While I believe it would be a cleaner solution, I don't see much benefit from it as the approach above is completely safe for developers and (also because) filesystem changes are thrown away with the end of container (GH runner) lifecycle.